pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -236,6 +236,28 @@ type SecuritySpec struct {
 	// +optional
 	// +nullable
 	KeyManagementService KeyManagementServiceSpec `json:"kms,omitempty"`
+	// OpenPolicyAgentService defines spec for the OPA service
+	// +optional
+	// +nullable
+	OpenPolicyAgentService *OpenPolicyAgentServiceSpec `json:"opa,omitempty"`
+}
+
+// OpenPolicyAgentServiceSpec contains options for communicating with the OPA server
+type OpenPolicyAgentServiceSpec struct {
+	// URL of the OPA service for the authentication
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	// +required
+	URL string `json:"url,omitempty"`
+	// Indicate whether the server certificate is validated by the client or not
+	// +optional
+	// +nullable
+	VerifySSL *bool `json:"verifySSL,omitempty"`
```

Comment:
Is the behavior of "not set" and "false" the same? If so, how about removing the pointer for simplicity? 
```suggestion
	VerifySSL bool `json:"verifySSL,omitempty"`
```